### PR TITLE
Amazon-AWS-Delete-Load-Balancer

### PR DESCRIPTION
**Pull request by mihaitusa**
Added op && unit test.
Signed off by Tusa Mihai: tusa.mihai@gmail.com

### CODE REVIEW

**Comment by mihaitusa** [comment]
@Bonczidai Please review it! 😄

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/utils/InputsUtil.java [hunk 1]
```diff
@@ -368,6 +375,16 @@ public static void validateKeyOrValueString(String input, boolean isKey) {
         }
     }
 
+    public static String getValidArn(String input) {
+        Pattern pattern = Pattern.compile("[a-zA-Z0-9\\:\\-\\/]+");
```
Comment (Tirla-Alin) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/utils/InputsUtil.java]: Isn't "\\\\:" a redundant escape? I think this can be reduced at this: "[a-zA-Z0-9:\\\\-/]+". Intellij should report this warning as well.

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/utils/InputsUtil.java [hunk 2]
```diff
@@ -368,6 +375,16 @@ public static void validateKeyOrValueString(String input, boolean isKey) {
         }
     }
 
+    public static String getValidArn(String input) {
```
Comment (Tirla-Alin) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/utils/InputsUtil.java]: I think this could be renamed to validateArn(String arn)

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/utils/InputsUtil.java [hunk 3]
```diff
@@ -368,6 +375,16 @@ public static void validateKeyOrValueString(String input, boolean isKey) {
...
+        Pattern pattern = Pattern.compile("[a-zA-Z0-9\\:\\-\\/]+");
+        if (!input.contains(COLON) || !input.startsWith(ARN) || !input.contains(AWS) || !input.contains(LOAD_BALANCING_API)
```
Comment (Tirla-Alin) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/utils/InputsUtil.java]: This if is hard to follow, also !input.contains(COLON) seems to be redundant IMHO because of countMatches(input, COLON) != 5. Also, can't we specify an ARN as arn:partition:service:region:account-id:resourcetype:resource (this has 6 colons)? My suggestion is to let the validation to the Amazon service.

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/loadbalancing/DeleteLoadBalancer.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    /**
+     * Deletes the specified Application Load Balancer and its attached listeners.
+     * Note: You can't delete a load balancer if deletion protection is enabled. If the load balancer does not exist or
+     * has already been deleted, the call succeeds. Deleting a load balancer does not affect its registered targets.
+     * For example, your EC2 instances continue to run and are still registered to their target groups. If you no longer
+     * need these EC2 instances, you can stop or terminate them.
+     *
+     * @param endpoint        Optional - Endpoint to which request will be sent.
+     *                        Example: "https://elasticloadbalancing.amazonaws.com"
+     *                        Default: "https://ec2.amazonaws.com"
```
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/loadbalancing/DeleteLoadBalancer.java]: No... the default endpoint is set to be ec2... Those op's are pointed to elasticloadbalancing ...

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/utils/InputsUtil.java (hunk 2 above)
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/utils/InputsUtil.java]: Ok

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/utils/InputsUtil.java (hunk 3 above)
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/utils/InputsUtil.java]: Ok

**Comment by mihaitusa** [comment]
@Tirla-Alin Modified all actions according to fall-back mechanism for endpoint... check it out!

**Comment by mihaitusa** [comment]
@Tirla-Alin That's all for my side... all yours now

**Comment by Bonczidai** [comment]
LGTM!

**Comment by mihaitusa** [comment]
@Bonczidai @Tirla-Alin Gentleman's ... just a couple of refactoring touch... tried to keep as clean as possible...
